ext/sodium/libsodium.c

```diff
@@ -1662,20 +1606,24 @@ PHP_FUNCTION(sodium_crypto_pwhash)
 		sodium_remove_param_values_from_backtrace(EG(exception));
 		RETURN_THROWS();
 	}
-	if (hash_len <= 0 || hash_len >= 0xffffffff) {
-		zend_throw_exception(sodium_exception_ce, "hash length must be greater than 0", 0);
+	if (hash_len <= 0) {
+		zend_argument_error(sodium_exception_ce, 1, "must be greater than 0");
+		RETURN_THROWS();
+	}
+	if (hash_len >= 0xffffffff) {
+		zend_argument_error(sodium_exception_ce, 1, "must be less than the maximum allowed value");
```

Comment:
Now that we are no longer restricted to positive messages, we can also use something like `XXX is too large` or `YYY is too long` here. I think that might be better for cases like this where we have a non-explicit upper bound.

Reply:
I absolutely agree! Actually, I've already changed a similar message to "XXX is too large" in my other PR, so I'll review these cases here as well shortly.